### ISSUE TITLE
cli: perhaps insecure-skip-tls-verify shouldn't be a global flag?

### ISSUE THREAD

**Issue by krancour** [akuity/kargo#633]
It seem to me that it would be enough to specify this once at login and then remember this.

On the one hand, it's a better user experience. On the other hand, it's less in-your-face about the issue.

But even a browser only asks you only once per session if you'd like to ignore cert error for a given site...

@jessesuen wdyt?

**Comment by krancour** [comment]
@gdsoumya tokens don't live forever, so periodic re-authentication is required and might naturally address that concern...

Except in cases where refresh tokens are supported...

I can just make it that refresh attempts do not happen in the event that a cert warning is currently being ignored. That would force the user to re-authenticate and actively choose once again to ignore the cert issue.

**Comment by gdsoumya** [comment]
What is the max ttl for the tokens kargo provides? If it's a very short period that's probably okay then btu if it's like a week or more idk I would rather not cache then.

**Comment by krancour** [comment]
We don't directly control it in all cases.

For admin tokens, it's configurable.

If you're using Dex, it's configurable.

If you're using OIDC with some other identity provider, it may or may not be configurable.

**Comment by jessesuen** [comment]
>It seem to me that it would be enough to specify this once at login and then remember this.

Are we talking about whether or not we should remember the `insecure-skip-tls-verify` preference in the `~/.config/kargo`? If so, then this should be fine. ~/.kube/config has a setting for this, as does Argo CD. Sometimes this is just the way folks need to operate.

**Comment by jessesuen** [comment]
deprioritizing since just a nice to have for 0.1.0.

**Comment by krancour** [comment]
> Are we talking about whether or not we should remember the insecure-skip-tls-verify preference in the ~/.config/kargo? If so, then this should be fine.

Yes.

> But the title of this is: "perhaps insecure-skip-tls-verify shouldn't be a global flag?"
>
> I would say it should be both

I'm fine with that.
